specification/batch/Azure.Batch/routes.tsp

```diff
@@ -0,0 +1,2151 @@
+  @projectedName("java", "disableNodeSchedulingInternal")
+  disableNodeScheduling is Ops.PostOperation<
+    MinimalMetadata & {
+      @doc("The ID of the Pool that contains the Compute Node.")
+      @path
+      poolId: string;
+
+      @doc("The ID of the Compute Node on which you want to disable Task scheduling.")
+      @path
+      nodeId: string;
+
+      @doc("The options to use for disabling scheduling on the Compute Node.")
+      @body
+      parameters?: BatchNodeDisableSchedulingParameters;
+    },
+    RequestSuccesResponseHeaders
+  >;
+
+  @summary("Enables Task scheduling on the specified Compute Node.")
+  @doc("""
+You can enable Task scheduling on a Compute Node only if its current scheduling
+state is disabled
+""")
+  @route("/pools/{poolId}/nodes/{nodeId}/enablescheduling")
+  @projectedName("java", "enableNodeSchedulingInternal")
+  enableNodeScheduling is Ops.PostOperation<
+    {
+      @doc("The ID of the Pool that contains the Compute Node.")
+      @path
+      poolId: string;
+
+      @doc("The ID of the Compute Node on which you want to enable Task scheduling.")
+      @path
+      nodeId: string;
+    },
+    RequestSuccesResponseHeaders
+  >;
+
+  @summary("Gets the settings required for remote login to a Compute Node.")
+  @doc("""
+Before you can remotely login to a Compute Node using the remote login
+settings, you must create a user Account on the Compute Node. This API can be
+invoked only on Pools created with the virtual machine configuration property.
+For Pools created with a cloud service configuration, see the GetRemoteDesktop
+API.
+""")
+  @route("/pools/{poolId}/nodes/{nodeId}/remoteloginsettings")
+  @projectedName("java", "getNodeRemoteLoginSettingsInternal")
+  getNodeRemoteLoginSettings is Ops.ReadOperation<
+    {
+      @doc("The ID of the Pool that contains the Compute Node.")
+      @path
+      poolId: string;
+
+      @doc("The ID of the Compute Node for which to obtain the remote login settings.")
+      @path
+      nodeId: string;
+    },
+    BatchNodeRemoteLoginSettings
+  >;
+
+  @summary("Gets the Remote Desktop Protocol file for the specified Compute Node.")
+  @doc("""
+Before you can access a Compute Node by using the RDP file, you must create a
+user Account on the Compute Node. This API can only be invoked on Pools created
+with a cloud service configuration. For Pools created with a virtual machine
+configuration, see the GetRemoteLoginSettings API.
+""")
+  @route("/pools/{poolId}/nodes/{nodeId}/rdp")
+  @projectedName("java", "getNodeRemoteDesktopFileInternal")
+  getNodeRemoteDesktopFile is Ops.ReadOperation<
+    {
+      @doc("The ID of the Pool that contains the Compute Node.")
+      @path
+      poolId: string;
+
+      @doc("""
+The ID of the Compute Node for which you want to get the Remote Desktop
+Protocol file.
+""")
+      @path
+      nodeId: string;
+    },
+    {
+      @body
+      @doc("A response containing the file content.")
+      file: bytes;
+    }
```

Comment:
If the response is `application/octet-stream`
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/batch/data-plane/Microsoft.Batch/stable/2023-11-01.18.0/BatchService.json#L10174

You must have the content-type defined in response e.g.
```ts
@header("content-type") contentType: "application/octet-stream";
```

`file: bytes` along is treated as bytes in JSON

Reply:
updated